Sources/MapboxMaps/Annotations/AnnotationOrchestrator.swift

```diff
@@ -106,15 +114,18 @@ public class AnnotationOrchestrator {
                                          style: style,
                                          view: view,
                                          mapFeatureQueryable: mapFeatureQueryable,
+                                         shouldPersist: shouldPersist,
                                          layerPosition: layerPosition)
     }
 
     /// Creates a `CircleAnnotationManager` which is used to manage a collection of `CircleAnnotation`s.
     /// - Parameters:
     ///   - id: Optional string identifier for this manager.
+    ///   - shouldPersist: Indicates whether the annotation layer should exist in after changing the map's style. Default value is `true`.
     ///   - layerPosition: Optionally set the `LayerPosition` of the layer managed.
     /// - Returns: An instance of `CircleAnnotationManager`
     public func makeCircleAnnotationManager(id: String = String(UUID().uuidString.prefix(5)),
+                                            shouldPersist: Bool = true,
```

Comment:
is this a public API? In Android we always use persistent layer, user can still remove the annotation even using the persistent layer.

Reply:
We didn't introduce any API breaking changes in Android, if we decide to introduce this flag, we need to align and add on both platforms.

Reply:
So this doesn't qualify as a breaking change on iOS since `shouldPersist` is defaulted to true.. Having said that, I'm having second thoughts on whether we should allow users the option of setting this to `false`.

@jmkiley : how do you feel about removing it from the public API (i.e. the `make*AnnotationManager` factory methods in `AnnotationOrchestrator`) ? We can leave them in the internal constructors of each of the manaagers..

Reply:
Because we provide a default value, I'm not sure that this would be a breaking change. Developers can still call `mapView.annotations.makePointAnnotationManager()` without passing in any values, for example, there is just an option to pass a value in for `shouldPersist` now.

Reply:
That said, we can remove this option to align with Android, but it seems like developers should have the option to disable this behavior for annotations.

Reply:
> So this doesn't qualify as a breaking change on iOS since shouldPersist is defaulted to true.. Having said that, I'm having second thoughts on whether we should allow users the option of setting this to false.

Sorry, yes, it is not a breaking change, but it's a new API that user could use to configure the annotations, and we don't expose such option on Android.

I'm not sure if we have a use case to set it to false, and considering API consistency, we don't have such options for LocationComponentPlugin as well, the default behaviour is that the location puck will persist. I think we should do the same for the Annotations.

Reply:
We can still add it to both platforms when customers request it, let's iterate on them 👍